sig-azure/0018-20180711-azure-availability-zones.md

```diff
@@ -0,0 +1,284 @@
+title: Azure Availability Zones
+authors:
+  - "@feiskyer"
+owning-sig: sig-azure
+participating-sigs:
+  - sig-azure
+  - sig-storage
+reviewers:
+  - name: "@khenidak"
+  - name: "@colemickens"
+approvers:
+  - name: "@brendanburns"
+editor: TBD
+creation-date: 2018-07-11
+last-updated: 2018-07-11
+status: provisional
+---
+
+# Azure Availability Zones
+
+## Table of Contents
+
+- [Azure Availability Zones](#azure-availability-zones)
+  - [Summary](#summary)
+  - [Scopes and Non-scopes](#scopes-and-non-scopes)
+    - [Scopes](#scopes)
+    - [Non-scopes](#non-scopes)
+  - [AZ label format](#az-label-format)
+  - [Cloud provider options](#cloud-provider-options)
+  - [Node registration](#node-registration)
+    - [Get by instance metadata](#get-by-instance-metadata)
+    - [Get by Go SDK](#get-by-go-sdk)
+  - [LoadBalancer and PublicIP](#loadbalancer-and-publicip)
+  - [AzureDisk](#azuredisk)
+    - [PVLabeler](#pvlabeler)
+    - [PersistentVolumeLabel](#persistentvolumelabel)
+    - [StorageClass](#storageclass)
+  - [Appendix](#appendix)
+
+## Summary
+
+This proposal aims to add [Azure Availability Zones (AZ)](https://azure.microsoft.com/en-us/global-infrastructure/availability-zones/) support to Kubernetes.
+
+## Scopes and Non-scopes
+
+### Scopes
+
+The proposal includes required changes to support availability zones for various functions in Azure cloud provider and AzureDisk volumes:
+
+- Detect availability zones automatically when registering new nodes (by kubelet or node controller) and node's label `failure-domain.beta.kubernetes.io/zone` will be replaced with AZ instead of fault domain
+- LoadBalancer and PublicIP will be provisioned with zone redundant
+- `GetLabelsForVolume` interface will be implemented for Azure managed disks so that PV label controller in cloud-controller-manager can appropriately add `Labels` and `NodeAffinity` to the Azure managed disk PVs. Additionally, `PersistentVolumeLabel` admission controller will be enhanced to achieve the same for Azure managed disks.
+- Azure Disk's `Provision()` function will be enhanced to take into account the zone of the node as well as `allowedTopologies` when determining the zone to create a disk in.
+
+> Note that unlike most cases, fault domain and availability zones mean different on Azure:
+>
+> - A Fault Domain (FD) is essentially a rack of servers. It consumes subsystems like network, power, cooling etc.
+> - Availability Zones are unique physical locations within an Azure region. Each zone is made up of one or more data centers equipped with independent power, cooling, and networking.
+>
+> An Availability Zone in an Azure region is a combination of a fault domain and an update domain (Same like FD, but for updates. When upgrading a deployment, it is carried out one update domain at a time). For example, if you create three or more VMs across three zones in an Azure region, your VMs are effectively distributed across three fault domains and three update domains.
+
```

Comment:
is there further classification like does an update domain have multiple fault domains ?

Reply:
No, each VM is assigned with a single update domain.